crates/fuel-core/src/service/adapters/graphql_api/off_chain.rs

```diff
@@ -250,37 +284,68 @@ impl OffChainDatabase for OffChainIterableKeyValueView {
                 }
             }
             Err(err) => iter::once(Err(err)).into_boxed(),
-        };
+        }
+    }
 
-        let non_base_asset_balance = self
-            .iter_all_filtered_keys::<CoinBalances, _>(Some(owner), None, Some(direction))
-            .filter_map(move |result| match result {
-                Ok(key) if *key.asset_id() != base_asset_id => Some(Ok(key)),
-                Ok(_) => None,
-                Err(err) => Some(Err(err)),
-            })
-            .map(move |result| {
-                result.and_then(|key| {
-                    let asset_id = key.asset_id();
-                    let coin_balance =
-                        self.storage_as_ref::<CoinBalances>()
-                            .get(&key)?
-                            .unwrap_or_default()
-                            .into_owned() as TotalBalanceAmount;
-                    Ok((*asset_id, coin_balance))
-                })
+    fn non_base_asset_balances<'a>(
+        &'a self,
+        owner: &Address,
+        start: Option<CoinBalancesKey>,
+        direction: IterDirection,
+        base_asset_id: &'a AssetId,
+    ) -> BoxedIter<'_, Result<(AssetId, u128), StorageError>> {
+        self.iter_all_filtered_keys::<CoinBalances, _>(
+            Some(owner),
+            start.as_ref(),
+            Some(direction),
+        )
+        .filter_map(move |result| match result {
```

Comment:
It would be valuable in the future to have a comment about this behavior because reading the comments on the review helped me but it's not really future proof and without it I wouldn't understand why `None` on base asset etc..

Reply:
I think it's quite clear, because the function name clearly says "NON_base_asset_balance", hence we map base assets to `None` to not process them further.

But of course I'm biased due to spending some significant time on this feature.

Can you suggest the message that would explain the exact thing that was confusing to you? I'm happy to include it for other readers.

Reply:
Mhhh yeah with the function name it's more understandable, didn't paid attention correctly when first read. I'm resolving